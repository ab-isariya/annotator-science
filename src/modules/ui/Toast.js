import {ToastContainer, Zoom} from 'react-toastify';

const Toast = () => {
  const typeClasses = {
    default: 'bg-grey-800',
    success: 'bg-blue-600',
    info: 'bg-grey-800',
    error: 'bg-red-600',
    warning: 'bg-orange-400',
    dark: 'bg-white-600 font-gray-300'
  };

  /**
   * Type comes from the toast call: toast.info is type=info
   * @param type
   * @return {string} - tailwind classes to apply
   */
  const toastTailwindClasses = ({type}) => {
    return (
      typeClasses[type || 'default'] +
      ' relative flex p-1 rounded-md justify-between overflow-hidden cursor-pointer'
    );
  };

  return (
    <ToastContainer
      toastClassName={toastTailwindClasses}
      className="p-0"
      position="bottom-center"
      transition={Zoom}
      progressClassName={'bg-holofoil'}
      // Notifications will be displayed as soon as a slot is free.
      limit={1}
      style={{width: 'auto'}}
      // Alerts include their own close button, and it works
      closeButton={false}
    />
  );
};

export default Toast;
