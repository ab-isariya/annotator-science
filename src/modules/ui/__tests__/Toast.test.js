import Toast from '../Toast';
import renderer from 'react-test-renderer';

describe('<Toast/>', () => {
  it('Renders without breaking', () => {
    const comp = renderer.create(<Toast />).toJSON();

    expect(comp).toMatchSnapshot();
  });
});
