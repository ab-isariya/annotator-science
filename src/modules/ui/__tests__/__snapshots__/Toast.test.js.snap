// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toast/> Renders without breaking 1`] = `
<div
  className="Toastify"
/>
`;
